process/smartContract/scDataGetter.go

```diff
@@ -0,0 +1,103 @@
+package smartContract
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/process"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/pkg/errors"
+)
+
+var maxGasValue = big.NewInt(math.MaxInt64)
+
+// scDataGetter can execute Get functions over SC to fetch stored values
+type scDataGetter struct {
+	vm       vmcommon.VMExecutionHandler
+	mutRunSc sync.Mutex
+}
+
+// NewSCDataGetter returns a new instance of scDataGetter
+func NewSCDataGetter(
+	vm vmcommon.VMExecutionHandler,
+) (*scDataGetter, error) {
+
+	if vm == nil {
+		return nil, process.ErrNoVM
+	}
+
+	return &scDataGetter{
+		vm: vm,
+	}, nil
+}
+
+// Get returns the value as byte slice of the invoked func
+func (scdg *scDataGetter) Get(scAddress []byte, funcName string, args ...[]byte) ([]byte, error) {
+	if scAddress == nil {
+		return nil, process.ErrNilScAddress
+	}
+	if len(funcName) == 0 {
+		return nil, process.ErrEmptyFunctionName
+	}
+
+	scdg.mutRunSc.Lock()
+	defer scdg.mutRunSc.Unlock()
+
+	vmInput := scdg.createVMCallInput(scAddress, funcName, args...)
+	vmOutput, err := scdg.vm.RunSmartContractCall(vmInput)
+	if err != nil {
+		return nil, err
+	}
+
+	// VM is formally verified and the output is correct
+	return scdg.processVMOutput(vmOutput)
+}
+
+func (scdg *scDataGetter) createVMCallInput(
+	scAddress []byte,
+	funcName string,
+	args ...[]byte,
+) *vmcommon.ContractCallInput {
+
+	argsInt := make([]*big.Int, 0)
+	for _, arg := range args {
+		argsInt = append(argsInt, big.NewInt(0).SetBytes(arg))
+	}
+
+	header := &vmcommon.SCCallHeader{
+		GasLimit:    big.NewInt(0),
+		Timestamp:   big.NewInt(0),
+		Beneficiary: big.NewInt(0),
+		Number:      big.NewInt(0),
+	}
+
+	vmInput := vmcommon.VMInput{
+		CallerAddr:  scAddress,
+		CallValue:   big.NewInt(0),
+		GasPrice:    big.NewInt(0),
+		GasProvided: maxGasValue,
+		Arguments:   argsInt,
+		Header:      header,
+	}
+	vmContractCallInput := &vmcommon.ContractCallInput{
+		RecipientAddr: scAddress,
+		Function:      funcName,
+		VMInput:       vmInput,
+	}
+
+	return vmContractCallInput
+}
+
+func (scdg *scDataGetter) processVMOutput(vmOutput *vmcommon.VMOutput) ([]byte, error) {
```

Comment:
change function name, as it does not process anything.

Reply:
changed to checkVMOutput